File: interviewbit-bit-manipulation-different-bits-sum-pairwise.py
class Solution:
    # @param A : list of integers
    # @return an integer
    def cntBits(self, A):
        bits = []
        for i in range(32):
            bit = 0
            for num in A:
                if num & 2 ** i:
                    bit += 1

            bits.append(bit)

        return sum((len(A) - bit) * bit * 2 for bit in bits) % (10 ** 9 + 7)

File: interviewbit-bit-manipulation-different-bits-sum-pairwise.java
public class Solution {
    public int cntBits(int[] A) {
        int[] bits = new int[31];
        for (int num : A) {
            for (int i = 0; i < 31; i++) {
                if ((num & (1 << i)) != 0) {
                    bits[i]++;
                }
            }
        }
        long diffBits = 0;
        for (int bit : bits) {
            diffBits += 2L * bit * (A.length - bit);
        }
        return (int) (diffBits % 1000000007);
    }
}
